src/include/detail/linalg/tdb_matrix.h

```diff
@@ -253,12 +257,25 @@ class tdbBlockedMatrix : public Matrix<T, LayoutPolicy, I> {
 #else
     auto data_ = std::unique_ptr<T[]>(new T[dimension * load_blocksize_]);
 #endif
-
-    Base::operator=(Base{std::move(data_), dimension, load_blocksize_});
+    if constexpr (std::is_same<MatrixBase, Matrix<T, LayoutPolicy, I>>::value) {
+      Base::operator=(Base{std::move(data_), dimension, load_blocksize_});
+    } else {
+      std::cout << "hit WithIDS case" << std::endl;
+#ifdef __cpp_lib_smart_ptr_for_overwrite
+      auto ids_data_ =
```

Comment:
No.

Reply:
This is not a useful comment, please explain more what you mean by this.